Source/WebCore/platform/graphics/gstreamer/GStreamerVideoSinkCommon.cpp

```diff
@@ -55,6 +55,10 @@ class WebKitVideoSinkProbe {
                     GST_DEBUG_OBJECT(pad, "FLUSH_START received, aborting all pending tasks in the player sinkTaskQueue.");
                     self->m_isFlushing = true;
                     player->sinkTaskQueue().startAborting();
+#if USE(GSTREAMER_GL)
+                    GST_DEBUG_OBJECT(pad, "Flushing current buffer in response to %" GST_PTR_FORMAT, info->data);
+                    player->flushCurrentBuffer();
```

Comment:
Have you seen the 3 FIXMEs couple lines below?

Also calling this leads to black frame rendered by the player until the next sample is received. Was this visually tested?

cc @ntrrgc

Reply:
Yes, I did.

I didn't see any blank frames so far in the rpi, at least.

Anyway, we have a problem with that dangling frame, somehow. I could investigate some more just in case it is something else, but we have a clear problem of buffer exhaustion.

Reply:
You shouldn't be seeing any blank buffers as `flushCurrentBuffer()` has some internal logic to replace the last buffer with a copy that doesn't use special memory.